src/main/groovy/org/labkey/gradle/plugin/TeamCity.groovy

```diff
@@ -229,8 +229,14 @@ class TeamCity extends Tomcat
                         task.dbPropertiesChanged = true
                         task.fn = {
                             properties.mergePropertiesFromFile()
-                            if (extension.dropDatabase)
-                                SqlUtils.dropDatabase(project, properties)
+                            if (extension.dropDatabase) {
+                                if (extension.getTeamCityProperty("testValidationOnly").equalsIgnoreCase("true")){
```

Comment:
I think you can do this without string comparison using a cast to a Boolean:

(Boolean) getTeamCityProperty("testValidationOnly")